### PR TITLE
moving more tfm dependent properties to targets file

### PR DESCRIPTION
**Pull request by Anipik**
Working Towards https://github.com/dotnet/runtime/issues/34525

### CODE REVIEW

**Comment by Anipik** [comment]
@ViktorHofer @ericstj can you take a look here ?

File: eng/references.props [hunk 1]
```diff
@@ -1,10 +0,0 @@
-<Project>
-  <PropertyGroup>
-    <!--
-      The Microsoft.NET.Sdk defaults the AssemblySearchPaths earlier, so default
-      the correct search paths here. We are rather restrictive by design - only ref assemblies should
-      be resolved by default.
-    -->
-    <AssemblySearchPaths>$(AssemblySearchPaths);$(RefPath);{RawFileName}</AssemblySearchPaths>
```
Comment (ericstj) [inline eng/references.props:8]: Why are we removing this?  IIRC this was done to avoid some undesireable probing locations like the GAC.

Comment (Anipik) [reply eng/references.props:8]: We already have this https://github.com/dotnet/runtime/blob/master/eng/references.targets#L5

**Review by ericstj — APPROVED** [review]

File: src/libraries/System.Text.Json/src/System.Text.Json.csproj [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <DocumentationFile>$(OutputPath)$(MSBuildProjectName).xml</DocumentationFile>
```
Comment (ahsonkhan) [inline src/libraries/System.Text.Json/src/System.Text.Json.csproj:4]: How is this property tfm dependent?

Comment (ViktorHofer) [reply src/libraries/System.Text.Json/src/System.Text.Json.csproj:4]: OutputPath is tfm dependent...

**Pull request merged**